src/neuroglancer/datasource/computed/tensorflow/frontend.ts

```diff
@@ -0,0 +1,278 @@
+  getComputation(config: any, {}, params: ComputedVolumeDataSourceParameters) {
+    let modelPath = verifyString(config['modelPath']);
+    if (!modelPath.endsWith('/')) {
+      modelPath += '/';
+    }
+
+    let model: import(/* webpackChunkName: "@tensorflow/tfjs" */ '@tensorflow/tfjs').InferenceModel;
+    let stdDev = 1.0;
+    let mean = 0.0;
+    let outputType = VolumeType.SEGMENTATION;
+
+    if (config['stdDev'] !== undefined) {
+      stdDev = verifyFloat(config['stdDev']);
+    }
+    if (config['mean'] !== undefined) {
+      mean = verifyFloat(config['mean']);
+    }
+    if (config['atype'] !== undefined) {
+      const atype = verifyString(config['atype']);
+      switch (atype) {
+        case 'classifier':
+          break;
+        case 'regressor':
+          outputType = VolumeType.IMAGE;
+          break;
+        default:
+          throw new Error(`Unknown algorithm type ${atype}. Must be "classifier" or "regressor"`);
+      }
+    }
+
+    // Load the model, then do a dummy inference run. This allows us to
+    // explicitly discover the output dimensions, and to compile the
+    // model on the gpu.
+    return loadTFjs()
+        .then(() => {
+          return tfjs.loadFrozenModel(
+              modelPath + 'tensorflowjs_model.pb', modelPath + 'weights_manifest.json');
+        })
+        .then(
+            (tfModel: import(/* webpackChunkName: "@tensorflow/tfjs" */ '@tensorflow/tfjs')
+                 .InferenceModel) => {
+              model = tfModel;
+              if (model.inputs.length !== 1) {
+                throw new Error('Only models with exactly one input are supported');
+              }
+              if (model.outputs.length !== 1) {
+                // Todo: support for multiple-output models.
+                throw new Error('Only models with exactly one output are supported');
+              }
+
+              // Create a blank tensor, run prediction, check output size
+              const dummyOutput = tfjs.tidy(() => {
+                const dummyInput:
+                    import(/* webpackChunkName: "@tensorflow/tfjs" */ '@tensorflow/tfjs').Tensor =
+                    tfjs.ones(<number[]>model.inputs[0].shape);
+                return <import(/* webpackChunkName: "@tensorflow/tfjs" */ '@tensorflow/tfjs')
+                            .Tensor>model.predict(dummyInput, {});
+              });
+
+              return dummyOutput.data().then(() => {
+                return dummyOutput;
+              });
+            })
+        .then(
+            (outputTensor:
+                 import(/* webpackChunkName: "@tensorflow/tfjs" */ '@tensorflow/tfjs').Tensor) => {
+              const inputShape = [1, 1, 1];
+              const outputShape = [1, 1, 1];
+              const inputTensor = model.inputs[0];
+              const inputDType = inputTensor.dtype;
+
+              let idx = 0;
+              for (let dim of inputTensor.shape!) {
```

Comment:
Perhaps validate these shapes a bit more?

Reply:
Added checks that the expected dimensionality of the model input and output is at most 3. In the future, it might be worth supporting models with undeclared input size, and to support batching along a configurable dimension. For now, and until those use cases are more clear, its easy enough to implement these operations in the model, prior to conversion.